__tests__/app.test.js

```diff
@@ -152,40 +152,65 @@ describe("/api/articles/:article_id", () => {
 });
 
 describe("/api/articles", () => {
-  test("GET 200: Responds with array of all article objects.", () => {
-    const exampleArticle = {
-      title: expect.any(String),
-      topic: expect.any(String),
-      author: expect.any(String),
-      votes: expect.any(Number),
-      created_at: expect.any(String),
-      article_img_url: expect.any(String),
-      article_id: expect.any(Number),
-      comment_count: expect.any(Number),
-    };
-    return request(app)
-      .get("/api/articles")
-      .expect(200)
-      .then(({ body }) => {
-        const { articles } = body;
-        expect(articles.length).toBe(13);
-        articles.forEach((article) => {
-          expect(article).toMatchObject(exampleArticle);
+  describe("GET /api/articles", () => {
+    test("GET 200: Responds with array of all article objects.", () => {
+      const exampleArticle = {
+        title: expect.any(String),
+        topic: expect.any(String),
+        author: expect.any(String),
+        votes: expect.any(Number),
+        created_at: expect.any(String),
+        article_img_url: expect.any(String),
+        article_id: expect.any(Number),
+        comment_count: expect.any(Number),
+      };
+      return request(app)
+        .get("/api/articles")
+        .expect(200)
+        .then(({ body }) => {
+          const { articles } = body;
+          expect(articles.length).toBe(13);
+          articles.forEach((article) => {
+            expect(article).toMatchObject(exampleArticle);
+          });
         });
-      });
+    });
+
+    test("GET 200: Responds with all article objects sorted by date in descending order", () => {
+      return request(app)
+        .get("/api/articles")
+        .expect(200)
+        .then(({ body }) => {
+          const { articles } = body;
+          expect(articles).toBeSortedBy("created_at", { descending: true });
+          articles.forEach((article) => {
+            expect(article).not.toHaveProperty("body");
+          });
+        });
+    });
   });
 
-  test("GET 200: Responds with all article objects sorted by date in descending order", () => {
-    return request(app)
-      .get("/api/articles")
-      .expect(200)
-      .then(({ body }) => {
-        const { articles } = body;
-        expect(articles).toBeSortedBy("created_at", { descending: true });
-        articles.forEach((article) => {
-          expect(article).not.toHaveProperty("body");
+  describe("GET /api/articles (topic query)", () => {
+    test("GET 200: Responds with array of all articles for given topic value.", () => {
+      return request(app)
+        .get("/api/articles?topic=cats")
+        .expect(200)
+        .then(({ body }) => {
+          const { articles } = body;
+          expect(articles.length).toBe(1);
+          articles.forEach((article) => {
```

Comment:
We maybe didn't need to worry about iteration if there is only one result here, or perhaps picking a topic with more articles would have made for a sturdier test